### PR TITLE
Add the wheel foreground view

### LINKED ISSUE #150: Create the WheelForegroundView
## Description
The [WheelForegroundView](https://github.com/toggl/mobileapp/blob/develop/Toggl.iOS/Views/EditDuration/WheelForegroundView.cs) does the following:
  - Draws the rainbow overlapping arcs between start end end times of a time entry
  - Draws the start/end time editing handles
  - Handles touch interaction
  - Provides interfaces to control the widget and its represented values

## Dependencies
Needs the Math on #147  
Depends on #148 

## Definition of done

- [x] The wheel foreground custom view is created

Drawing:

- [x] It draws start end end editing handles for stopped time entries
- [x] It draws only the start editing handle for running time entries, drawing the end of the wheel arc as "now"
- [x] It draws rainbow coloured arcs/wheels based on the number of hours loops (1 loop = 1 hour)
- [x] It considers dark mode

Touch interaction:

- [x] It handles touch interaction when moving the editing handles
- [x] It handles touch interaction when moving the wheel arcs
- [x] It provides haptic feedback when handling the handles/arcs for every minute swirled

Interface:

- [x] It has configurable minimum and maximum start end end times
- [x] It provides an interface to configure the wheel as in "running mode" to represent running time entries
- [x] It provides an interface to set the start time
- [x] It provides an interface to set the end time
- [x] It emits events when start/end times are changed due to user interaction

### PR DESCRIPTION
**Pull request by juanlaube**
## Info
This adds the wheel foreground view.

### Summary
This adds the wheel foreground view, the implementation is a port of the same view in the old app and I took the opportunity to make things more swifty.

This was initially started by @heytherewill and I took over.

### Technical considerations
Not much to add here, most of the implementation details is a port of the old app. In this case I didn't add an overservable that emits when the values are changed, instead I made the view a subclass of `UIControl` and send a `.valueChanged` action when the start and/or end times are changed. This can be used with RxCocoa after.

### Relationships
Closes #150 

### Pull Request checklist
- [x] The code does not hinder the app's performance
- [x] The code does not introduce regressions
- [x] The code in this PR is tested thoroughly enough that changing the behavior to an incorrect one breaks tests
- [ ] The code used custom templates

### Pointers
I would consider comparing the implementation with the old app since it's quite similar.
I recommend also testing on an iPhone because of the feedback.

### Review template

- [ ] The code is thoroughly tested in well written and resilient manner
- [ ] The code is performant and leads to an optimal user experience
- [ ] The code is testable and does not cut corners for the sake of delivering
- [ ] The code considers older platforms well and will work in older devices just like it will in new ones

- [ ] The code considers light and dark theme
- [ ] The code considers RTL and does not mention left/right
- [ ] The code considers i18n
  - [ ] There are no hardcoded strings in the UI
  - [ ] All strings are included in every language we support OR there's an issue requesting new translations
- [ ] There are no hard-coded numbers referring to dimensions or font sizes
- [ ] The code considers UI elements that are similar and extracts their similarities into styles to avoid duplication

## Merge permissions

🦑

### CODE REVIEW

File: Modules/Timer/Timer/StartEdit/Views/Wheel/Cap.swift [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+    }
+    // swiftlint:enable identifier_name
+    
+    public static func endCap(scale: (CGFloat) -> (CGFloat)) -> Cap {
+        // ASSETS: Replace icon
+        let icon: UIImage
+        if #available(iOS 13, *) {
+            icon = UIImage(systemName: "stop.fill")!
+        } else {
+            icon = UIImage(imageLiteralResourceName: "icEndLabel")
```
[the remark this answers is not in the document]
Comment (juanlaube) [reply Modules/Timer/Timer/StartEdit/Views/Wheel/Cap.swift]: Fixed, I added the assets.

**Comment by juanlaube** [comment]
@Odrakir take a look again, the glitch was caused by a bug I introduced when refactoring the math to round to the nearest minute, I was adding seconds instead of subtracting 🙈

**Comment by Odrakir** [comment]
There's something else I found in the functional testing, and I'm not sure about: Shouldn't the limit of the start/end handles be the current position of the other handle and not the initial position? At least in the current app that's what happens, but maybe in this case this is waiting to be wired with the model.

**Comment by juanlaube** [comment]
Yes, that part specifically is not wired to the model yet, by default the minstart is .distantPast, the maxStart is now, the minEnd is now and the maxEnd is .distantFuture, and isRunning is false.

**Comment by daividssilverio** [comment]
There's an issue to set the right max/min start/end times #229 .

**Comment by juanlaube** [comment]
@sofokli1 please take a look again, the glitch was caused by a rounding issue.

**Pull request merged**